tools/toktx/toktx.cc

```diff
@@ -1257,6 +1291,39 @@ toktxApp::main(int argc, _TCHAR *argv[])
                 exitCode = 2;
                 goto cleanup;
             }
+        } else if (options.astc) {
+            commandOptions::astcOptions& astcopts = options.astcopts;
+#if TRAVIS_DEBUG
+            astcopts.print();
+#endif
+            if (options.inputSwizzle.size()) {
+                for (i = 0; i < 4; i++) {
```

Comment:
This and defaultSwizzle below should probably be `i < options.inputSwizzle.size()` just in case somebody messes with the strings. `size()` is validated to be 4 when the option is set so perhaps I am being over cautious. What do you think?

Reply:
I think it makes sense. If we have less than 4 characters in the `inputSwizzle` the `astcopts.inputSwizzle` will be invalid as well. But this is probably future proof incase the size changes.

Fixed in https://github.com/KhronosGroup/KTX-Software/pull/433/commits/8f42128b3c643db8b980d31167f8e9f34e1506e8